### ISSUE TITLE
Accessing data via the API - mio integration

### ISSUE THREAD

**Issue by lcantrell79** [reapit/foundations#8076]
**Describe the bug**
Our mutual customer Urban and Rural (Customer id: UAR-GBDF) have raised that there are some properties solicitors not updating via the integration.

I used Postman to access the data to analyse the problem. 

1 property I found the buyer solicitor id missing from the conveyancing records via the API but are visible to the users in Reapit, please advise why this is the case.

![image](https://user-images.githubusercontent.com/93535875/199987203-c812acec-98e0-497c-b6c7-09b8d74508fe.png)
![image](https://user-images.githubusercontent.com/93535875/199987224-adcd8946-1ffa-468b-9c0c-c1a9f9e520a7.png)

1 property has a buyer solicitor id but when I try access the company it says "not found"

![image](https://user-images.githubusercontent.com/93535875/199987083-7fa90888-4905-46ca-bde4-2e2452f2ea33.png)
![image](https://user-images.githubusercontent.com/93535875/199987116-d5e49f91-50bd-4f3d-af74-64bb3bfed8bc.png)
![image](https://user-images.githubusercontent.com/93535875/199987153-d29ea341-8a3b-4d86-a27a-a4808ea683b6.png)

**Comment by AshDeeming** [comment]
Hi @lcantrell79 
In the above example The buyer was archived via AgencyCloud on 1st Nov 2022 as the sale appears to be from June 2022.

**Comment by lcantrell79** [comment]
Hi @AshDeeming 
Would the agent have archived the buyer? because the agent is still progressing this sale and they are querying with us why the information is not pulling through to mio via the integration.

**Comment by AshDeeming** [comment]
Hi @lcantrell79 
Apologies for any confusion, I meant that their Applicant record was made inactive - this may be part of the agents process however could impact the return depending on the call

**Comment by AshDeeming** [comment]
Thanks @lcantrell79 
Please see the below;
Buyer id - BED22000800 - Applicant ID: BED220831 - applicant archived
![image](https://user-images.githubusercontent.com/61687775/200344658-3c9d4600-69ee-4a88-adfc-420a8facfd1b.png)
& no details populated on the Accepted Offer (
![image](https://user-images.githubusercontent.com/61687775/200344447-c319f967-0a9b-4543-a1e9-e254019b4bca.png)

Buyer id - BED22001591 
Applicant ID BED221549 - Is Inactive:
![image](https://user-images.githubusercontent.com/61687775/200345270-a7e41b22-7f75-4c25-87ae-a6b05a662734.png)
& no solicitor details populated on Accepted Offer;
![image](https://user-images.githubusercontent.com/61687775/200345470-4d0c4ae1-ba24-450f-93c2-6732cbddc515.png)

**Comment by lcantrell79** [comment]
Thank you @AshDeeming, I'll see what's happening in the agents process

**Comment by lcantrell79** [comment]
@AshDeeming I'm waiting on feedback from the agents process.
But I wanted to ask why the screenshots you provide are from the "Buyer and sales details" tab when this tab is only populated later in the process, on Sandbox it's usually when the status changes to Exchanged. 
While at Sale Agreed status I normally view the "Offers and Reservations" tab to see if the Solicitors have been added, would this still be considered a way to view the solicitors information?

**Comment by plittlewood-rpt** [comment]
Hi @lcantrell79 I think there might be a few different things intermingled in this item so to try and just confirm my understanding:

1. You have encountered an issue where you have called the Conveyancing API where you have received a response where the buyerSolicitorId is NOT set, but appears to be set in AgencyCloud
2. For some results from the Conveyancing service, you are unable to lookup the company from `GET /companies/{id}/` where id is the value of the `buyerSolicitorId` from the conveyancing endpoint response.
3. You queried the screenshots provided earlier by Ash. I think we can discount this. The conveyancing service looks at the information from the Offers and Reservations tab (and Sales Progress screen which shows the same information with regard to Solicitors). You are right that the Buyer & Sales Details screen is populated later in the process. Apologies for any confusion here

If the above is correct (please let me know if not), my responses to 1/2 are as follows:

1. I've had a look at the first example you gave (Mr Jamie Robinson against property BED220139) and the buyerSolicitorId is populated. The only thing I can think may have happened here is that some data in the database was missing that the API expects to be there. I can see that a link between his applicant record and chosen solicitor was amended today. Can you let me know if you have any current examples of this problem so I can take another look? 
2. We recently added security to the companies service  as part of #7586 that is applied as per other services when an app is installed for an office group rather than a whole company. This looks to be the case based on the client id (UAR-GBDF) you provided. The solicitor company associated with property BED220378 and buyerId BED22001591 is assigned to the Newport Pagnell office. As a result, if this office is NOT part of the BDF office group, a user hitting the API as part of the group would not be able to retrieve the company record.

I'd be grateful if you could review the above and come back to me.

Thanks

**Comment by lcantrell79** [comment]
@plittlewood-rpt 
Yes your points 1 to 3 are correct. Thank you for clearing up the access for office groups when accessing companies, I thought that may be the case.

Your response 1, I have one more example provided by the agent, I think they have amended most of the others today.
 "propertyId": "BED220227"  :  "buyerId": "BED22000365",   :     "buyerSolicitorId": null,

**Comment by lcantrell79** [comment]
> 2\. The solicitor company associated with property BED220378 and buyerId BED22001591 is assigned to the Newport Pagnell office.

Hi @plittlewood-rpt 
A follow up question if it's OK.
I assumed users would know if the companies are assigned to other offices, but some do not.
 
How would a user assign a company to an office when creating a company?
How could a user change the office a company is assigned to after the company has been created?

This information would help us immensely to support the Reapit integration users more efficiently.

**Comment by plittlewood-rpt** [comment]
Hi @lcantrell79 - relationships between a company and an office are managed using the Contact details -> Negotiators/Office panel on the Company edit screen in AgencyCloud. Hope this helps

**Comment by lcantrell79** [comment]
> Hi @lcantrell79 - relationships between a company and an office are managed using the Contact details -> Negotiators/Office panel on the Company edit screen in AgencyCloud. Hope this helps

Hi, The customer has done this for Property BED220386 and company BIG20001006 as below but it is still not allowing us access to the company
![reapit screenshot_](https://user-images.githubusercontent.com/93535875/215081793-4595bd50-683e-4999-9bef-ba264a3752cb.jpg)

When I run a query in postman (because mio is not updating for the customer :) ) then I get this message. Please help.
![image](https://user-images.githubusercontent.com/93535875/215082007-83b12d9e-f0b6-4783-872b-3023af3d6106.png)

**Comment by plittlewood-rpt** [comment]
Hi @lcantrell79 I've just had a quick look and it's happening because the company is not primarily owned by the BED office which is the only office defined by the group UAR-GBDF. It is however a shared office as I saw from your screenshot (which I've removed as it contained PII - please be mindful that this is a public repository - thanks). There is a bug in the filtering logic in this API which isn't handling shared ownership correctly which I'll get sorted now for you

**Comment by lcantrell79** [comment]
@plittlewood-rpt Apologies I will be mindful in future. Thank you for addressing the issue so swiftly.

**Comment by plittlewood-rpt** [comment]
@lcantrell79 this is now resolved. Many thanks
